File: cpg-solidity/src/main/java/de/fraunhofer/aisec/cpg/frontends/solidity/nodes/Require.java
package de.fraunhofer.aisec.cpg.frontends.solidity.nodes;

import de.fraunhofer.aisec.cpg.graph.SubGraph;
import de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression;
import org.jetbrains.annotations.Nullable;
import org.neo4j.ogm.annotation.Relationship;

public class Require extends Expression {
    @Relationship(value = "CONDITION", direction = "OUTGOING")
    @SubGraph("AST")
    @Nullable
    private Expression condition;

    @Relationship(value = "MESSAGE",direction = "OUTGOING")
    @SubGraph("AST")
    @Nullable
    private Expression message;

    @Nullable
    public final Expression getCondition() {
        return this.condition;
    }

    public final void setCondition(@Nullable Expression var1) {
        this.condition = var1;
    }


    @Nullable
    public final Expression getMessage() {
        return this.message;
    }

    public final void setMessage(@Nullable Expression var1) {
        this.message = var1;
    }
}

File: cpg-solidity/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/solidity/nodes/Require.kt
package de.fraunhofer.aisec.cpg.frontends.solidity.nodes

import de.fraunhofer.aisec.cpg.graph.AST
import de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression
import org.neo4j.ogm.annotation.Relationship

class Require : Expression() {
    @field:Relationship(value = "CONDITION", direction = Relationship.Direction.OUTGOING)
    @AST
    var condition: Expression? = null
        set(var1) {
            field = var1
            condition!!.addNextDFG(this)
        }

    @field:Relationship(value = "MESSAGE", direction = Relationship.Direction.OUTGOING)
    @AST
    var message: Expression? = null
        set(var1) {
            field = var1
            message!!.addNextDFG(this)
        }
}